main.py

```
@@ -81,38 +85,82 @@ def run(self):
+                with open(self._dest_path, "wb") as f:
+                    while True:
+                        if self.isInterruptionRequested():
+                            return
+                        buf = resp.read(65536)
+                        if not buf:
+                            break
+                        f.write(buf)
+                        done += len(buf)
+                        self.progress.emit(done, total)
+            self.finished.emit(self._dest_path)
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)